base/stream.jl

```diff
@@ -1063,10 +1063,13 @@ function readuntil(x::LibuvStream, c::UInt8; keep::Bool=false)
     return bytes
 end
 
-uv_write(s::LibuvStream, p::Vector{UInt8}) = GC.@preserve p uv_write(s, pointer(p), UInt(sizeof(p)))
+uv_write(s::LibuvStream, p::Vector{UInt8}) = GC.@preserve p uv_write(s, pointer(p), sizeof(p) % UInt64) # UInt64 is safe here, also for 32-bit, then get UInt32 back from uv_write
 
 # caller must have acquired the iolock
 function uv_write(s::LibuvStream, p::Ptr{UInt8}, n::UInt)
+    if Int == Int32 && n isa UInt32 && reinterpret(Int32, i) < 0 # aka n >= typemax(Int32) on 32-bit, no-op on 64-bit
+        throw(ArgumentError(LazyString("On 32-bit strings larger than 2 GB can't be printed in one go! Iterating over strings and printing the Chars is always safe, but splitting string into two SubStrings and printing sepratelyly may not be.")))
```

Comment:
This error text is problematic. Note that we are inside `uv_write` which is not just for *printing strings* but writing out memory buffers in a more general setting. There are also typos in there.

I am not yet convinced we want this exception, but if we do, maybe the message should be more on point, e.g.
```suggestion
        throw(ArgumentError(LazyString("cannot write more than 2 GB at a time")))
```
or so